### PR TITLE
recycler view room button

### PR DESCRIPTION
**Pull request by vdcn1**
colocando botao de entrar na sala em cada item do recycler view

### CODE REVIEW

File: app/src/main/java/com/example/deejai/RoomSelection/RoomAdapter.kt [hunk 1]
```diff
@@ -1,5 +1,8 @@
 package com.example.deejai.RoomSelection
 
+import android.content.Context
+import android.content.Intent
+import android.util.Log
```
Comment (lcc) [inline app/src/main/java/com/example/deejai/RoomSelection/RoomAdapter.kt]: If this was a debugging log, please remove it.

File: app/src/main/java/com/example/deejai/RoomSelection/RoomAdapter.kt [hunk 2]
```diff
@@ -37,6 +41,15 @@ class RoomAdapter(val rooms : List<Room>) : Adapter<RoomAdapter.RoomViewHolder>(
             roomName = itemView.findViewById(R.id.room_name)
             roomDistance = itemView.findViewById(R.id.room_distance)
             roomImage = itemView.findViewById(R.id.room_image) as ImageView
+            itemView.setOnClickListener(this)
+        }
+
+        override fun onClick(v: View?) {
+            val position = adapterPosition
```
Comment (lcc) [inline app/src/main/java/com/example/deejai/RoomSelection/RoomAdapter.kt]: This value is not being used, please remove it.

File: app/src/main/java/com/example/deejai/RoomSelection/RoomAdapter.kt [hunk 3]
```diff
@@ -8,9 +11,11 @@ import androidx.cardview.widget.CardView
 import androidx.recyclerview.widget.RecyclerView.ViewHolder
 import androidx.recyclerview.widget.RecyclerView.Adapter
 import android.view.LayoutInflater
+import androidx.core.content.ContextCompat.startActivity
+import androidx.recyclerview.widget.RecyclerView
 import com.example.deejai.R
 
-class RoomAdapter(val rooms : List<Room>) : Adapter<RoomAdapter.RoomViewHolder>() {
+class RoomAdapter(private val rooms : List<Room>, private val c : Context) : Adapter<RoomAdapter.RoomViewHolder>() {
```
Comment (lcc) [inline app/src/main/java/com/example/deejai/RoomSelection/RoomAdapter.kt]: Rename this context to something a bit more explanative (e.g.: context).

File: app/src/main/java/com/example/deejai/RoomSelection/RoomAdapter.kt [hunk 4]
```diff
@@ -37,6 +41,15 @@ class RoomAdapter(val rooms : List<Room>) : Adapter<RoomAdapter.RoomViewHolder>(
             roomName = itemView.findViewById(R.id.room_name)
             roomDistance = itemView.findViewById(R.id.room_distance)
             roomImage = itemView.findViewById(R.id.room_image) as ImageView
+            itemView.setOnClickListener(this)
+        }
+
+        override fun onClick(v: View?) {
+            val position = adapterPosition
+            val myintent = Intent(c, RoomPlayer::class.java)
```
Comment (lcc) [inline app/src/main/java/com/example/deejai/RoomSelection/RoomAdapter.kt]: There is no need to name the val myintent, intent sends the same message and is smaller. Please change it.

File: app/src/main/java/com/example/deejai/RoomSelection/RoomAdapter.kt [hunk 5]
```diff
@@ -37,6 +41,15 @@ class RoomAdapter(val rooms : List<Room>) : Adapter<RoomAdapter.RoomViewHolder>(
...
+            Log.d("adapter", roomName.text.toString())
```
Comment (lcc) [inline app/src/main/java/com/example/deejai/RoomSelection/RoomAdapter.kt]: Is this a debugging log? Please remove it.

File: app/src/main/java/com/example/deejai/RoomSelection/RoomAdapter.kt [hunk 6]
```diff
@@ -37,6 +41,15 @@ class RoomAdapter(val rooms : List<Room>) : Adapter<RoomAdapter.RoomViewHolder>(
...
+            myintent.putExtra("roomName", roomName.text.toString())
```
Comment (lcc) [inline app/src/main/java/com/example/deejai/RoomSelection/RoomAdapter.kt]: Put this string in a constant.

File: app/src/main/res/layout/activity_room_player.xml [hunk 7]
```diff
@@ -0,0 +1,24 @@
...
+        android:layout_marginTop="270dp"
+        android:layout_marginEnd="176dp"
+        android:layout_marginRight="176dp"
+        android:layout_marginBottom="442dp"
+        android:text="TextView"
+        app:layout_constraintBottom_toBottomOf="parent"
+        app:layout_constraintEnd_toEndOf="parent"
+        app:layout_constraintStart_toStartOf="parent"
+        app:layout_constraintTop_toTopOf="parent" />
+</androidx.constraintlayout.widget.ConstraintLayout>
```
Comment (lcc) [inline app/src/main/res/layout/activity_room_player.xml]: Add a newline to the end of the file please.

File: app/src/main/java/com/example/deejai/RoomSelection/RoomPlayer.kt [hunk 8]
```diff
@@ -0,0 +1,21 @@
...
+import android.widget.TextView
+import android.widget.Toast
+import com.example.deejai.R
+
+class RoomPlayer : AppCompatActivity() {
+
+    override fun onCreate(savedInstanceState: Bundle?) {
+        super.onCreate(savedInstanceState)
+        setContentView(R.layout.activity_room_player)
+        var bundle :Bundle ?=intent.extras
```
Comment (lcc) [inline app/src/main/java/com/example/deejai/RoomSelection/RoomPlayer.kt]: Please fix this identation. Should be something like:
         var bundle : Bundle? = intent.extras

File: app/src/main/java/com/example/deejai/RoomSelection/RoomPlayer.kt [hunk 9]
```diff
@@ -0,0 +1,21 @@
...
+        var message = bundle!!.getString("roomName") // 1
```
Comment (lcc) [inline app/src/main/java/com/example/deejai/RoomSelection/RoomPlayer.kt]: Remove the comment, it doesn't help to understand the code.

File: app/src/main/java/com/example/deejai/RoomSelection/RoomPlayer.kt [hunk 10]
```diff
@@ -0,0 +1,21 @@
...
+        Toast.makeText(this, message, Toast.LENGTH_SHORT).show()
```
Comment (lcc) [inline app/src/main/java/com/example/deejai/RoomSelection/RoomPlayer.kt]: Remove this toast as it adds nothing to the resulting app.

File: app/src/main/java/com/example/deejai/RoomSelection/RoomPlayer.kt [hunk 11]
```diff
@@ -0,0 +1,21 @@
...
+
```
Comment (lcc) [inline app/src/main/java/com/example/deejai/RoomSelection/RoomPlayer.kt]: Remove this empty line.

File: app/src/main/java/com/example/deejai/RoomSelection/RoomPlayer.kt [hunk 12]
```diff
@@ -0,0 +1,21 @@
...
+        val mytext : TextView = findViewById(R.id.room_name)
+        mytext.setText(message)
```
Comment (lcc) [inline app/src/main/java/com/example/deejai/RoomSelection/RoomPlayer.kt]: You can use mytext.text = message.

**Review by lcc — COMMENTED** [review]
This commit needs minor fixes, mostly logs and variable naming, please address them for this PR to be approved.

File: app/src/main/java/com/example/deejai/RoomSelection/RoomAdapter.kt (hunk 1 above)
Comment (vdcn1) [reply app/src/main/java/com/example/deejai/RoomSelection/RoomAdapter.kt]: done

File: app/src/main/java/com/example/deejai/RoomSelection/RoomAdapter.kt (hunk 2 above)
Comment (vdcn1) [reply app/src/main/java/com/example/deejai/RoomSelection/RoomAdapter.kt]: done

File: app/src/main/java/com/example/deejai/RoomSelection/RoomAdapter.kt (hunk 3 above)
Comment (vdcn1) [reply app/src/main/java/com/example/deejai/RoomSelection/RoomAdapter.kt]: done

File: app/src/main/java/com/example/deejai/RoomSelection/RoomAdapter.kt (hunk 4 above)
Comment (vdcn1) [reply app/src/main/java/com/example/deejai/RoomSelection/RoomAdapter.kt]: done

File: app/src/main/java/com/example/deejai/RoomSelection/RoomAdapter.kt (hunk 5 above)
Comment (vdcn1) [reply app/src/main/java/com/example/deejai/RoomSelection/RoomAdapter.kt]: done

File: app/src/main/java/com/example/deejai/RoomSelection/RoomAdapter.kt (hunk 6 above)
Comment (vdcn1) [reply app/src/main/java/com/example/deejai/RoomSelection/RoomAdapter.kt]: done

File: app/src/main/res/layout/activity_room_player.xml (hunk 7 above)
Comment (vdcn1) [reply app/src/main/res/layout/activity_room_player.xml]: done

File: app/src/main/java/com/example/deejai/RoomSelection/RoomPlayer.kt (hunk 8 above)
Comment (vdcn1) [reply app/src/main/java/com/example/deejai/RoomSelection/RoomPlayer.kt]: done

File: app/src/main/java/com/example/deejai/RoomSelection/RoomPlayer.kt (hunk 9 above)
Comment (vdcn1) [reply app/src/main/java/com/example/deejai/RoomSelection/RoomPlayer.kt]: done

File: app/src/main/java/com/example/deejai/RoomSelection/RoomPlayer.kt (hunk 10 above)
Comment (vdcn1) [reply app/src/main/java/com/example/deejai/RoomSelection/RoomPlayer.kt]: done

File: app/src/main/java/com/example/deejai/RoomSelection/RoomPlayer.kt (hunk 11 above)
Comment (vdcn1) [reply app/src/main/java/com/example/deejai/RoomSelection/RoomPlayer.kt]: done

File: app/src/main/java/com/example/deejai/RoomSelection/RoomPlayer.kt (hunk 12 above)
Comment (vdcn1) [reply app/src/main/java/com/example/deejai/RoomSelection/RoomPlayer.kt]: done

**Review by lcc — APPROVED** [review]

**Pull request merged**